runtime/lua/vim/treesitter/dev.lua

```diff
@@ -501,12 +501,19 @@ function M.inspect_tree(opts)
     end,
   })
 
-  api.nvim_create_autocmd('BufHidden', {
+  api.nvim_create_autocmd({ 'WinClosed' }, {
     group = group,
     buffer = buf,
-    once = true,
     callback = function()
-      close_win(w)
+      local windows = vim.fn.win_findbuf(buf) --[=[@as integer[]]=]
+      if #windows == 1 then
+        close_win(w)
+        return true
```

Comment:
This seems weird. Why is the WinClosed handler closing the window? Seems redundant.

Also why is the window being stored in both `w` and `win`? https://github.com/justinmk/neovim/blob/52d2851ca4e747146c30e1996c08bf504d99fe95/runtime/lua/vim/treesitter/dev.lua#L329-L340

What is the difference between `w` and `win` ?

Reply:
> * `win` and `buf` for the buffer that has the code
> * `w` and `b` for the InspectTree buffer

for now let's add docstrings on those. can use that exact wording